MassSpectrometry/MzSpectra/SpectralSimilarity.cs

```diff
@@ -161,8 +176,12 @@ private double[] NormalizeSpectrumSum(double[] spectrum)
         #region similarityMethods
 
         //The cosine similarity returns values between 1 and -1 with 1 being closes and -1 being opposite and 0 being orthoganal
-        public double CosineSimilarity()
+        public double? CosineSimilarity()
```

Comment:
do we really want null?

Reply:
we don't want null but it happens when all mz are less than 300 in theoretical(library) peaks

Reply:
I mean it happens that there is no intensity pair when all mz are less than 300 in theoretical(library) peaks. I need to assign a value to it